<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.27">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Mateo Jiménez Sotelo">
<meta name="dcterms.date" content="2026-02-01">

<title>RNA-seq Analysis from recount3</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="report_files/libs/clipboard/clipboard.min.js"></script>
<script src="report_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="report_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="report_files/libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="report_files/libs/quarto-html/popper.min.js"></script>
<script src="report_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="report_files/libs/quarto-html/anchor.min.js"></script>
<link href="report_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="report_files/libs/quarto-html/quarto-syntax-highlighting-ed96de9b727972fe78a7b5d16c58bf87.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="report_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="report_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="report_files/libs/bootstrap/bootstrap-0cf178164be81c6ad854e95b21275c80.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN" && texText && texText.data) {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#rna-seq-analysis-from-recount3" id="toc-rna-seq-analysis-from-recount3" class="nav-link active" data-scroll-target="#rna-seq-analysis-from-recount3"><span class="header-section-number">1</span> RNA-seq Analysis from recount3</a>
  <ul class="collapse">
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction"><span class="header-section-number">1.1</span> Introduction</a></li>
  <li><a href="#uploading-data" id="toc-uploading-data" class="nav-link" data-scroll-target="#uploading-data"><span class="header-section-number">1.2</span> Uploading data</a>
  <ul class="collapse">
  <li><a href="#libraries" id="toc-libraries" class="nav-link" data-scroll-target="#libraries"><span class="header-section-number">1.2.1</span> Libraries</a></li>
  <li><a href="#data-access" id="toc-data-access" class="nav-link" data-scroll-target="#data-access"><span class="header-section-number">1.2.2</span> Data access</a></li>
  </ul></li>
  <li><a href="#exploration" id="toc-exploration" class="nav-link" data-scroll-target="#exploration"><span class="header-section-number">1.3</span> Exploration</a>
  <ul class="collapse">
  <li><a href="#metadata" id="toc-metadata" class="nav-link" data-scroll-target="#metadata"><span class="header-section-number">1.3.1</span> Metadata</a></li>
  <li><a href="#format" id="toc-format" class="nav-link" data-scroll-target="#format"><span class="header-section-number">1.3.2</span> Format</a></li>
  </ul></li>
  <li><a href="#preprocessing" id="toc-preprocessing" class="nav-link" data-scroll-target="#preprocessing"><span class="header-section-number">1.4</span> Preprocessing</a>
  <ul class="collapse">
  <li><a href="#clustering-samples-into-their-id-groups" id="toc-clustering-samples-into-their-id-groups" class="nav-link" data-scroll-target="#clustering-samples-into-their-id-groups"><span class="header-section-number">1.4.1</span> Clustering samples into their ID groups</a></li>
  <li><a href="#basic-qc" id="toc-basic-qc" class="nav-link" data-scroll-target="#basic-qc"><span class="header-section-number">1.4.2</span> Basic QC</a></li>
  <li><a href="#drop-lowly-expressed-genes" id="toc-drop-lowly-expressed-genes" class="nav-link" data-scroll-target="#drop-lowly-expressed-genes"><span class="header-section-number">1.4.3</span> Drop lowly expressed genes</a></li>
  <li><a href="#normalization" id="toc-normalization" class="nav-link" data-scroll-target="#normalization"><span class="header-section-number">1.4.4</span> Normalization</a></li>
  </ul></li>
  <li><a href="#resultados" id="toc-resultados" class="nav-link" data-scroll-target="#resultados"><span class="header-section-number">1.5</span> Resultados</a></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results"><span class="header-section-number">1.6</span> Results</a></li>
  </ul></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references"><span class="header-section-number">2</span> References</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">RNA-seq Analysis from recount3</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Mateo Jiménez Sotelo </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 1, 2026</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<div class="text-left">
<section id="rna-seq-course-project" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="rna-seq-course-project">RNA-seq course project</h3>
</section>
<section id="lcg22" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="lcg22">LCG22</h3>
</section>
<section id="unam" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="unam">UNAM</h3>
<hr>
<p><strong>Dataset:</strong> SRP068976<br>
<strong>Platform:</strong> recount3</p>
</section>
</div>
<section id="rna-seq-analysis-from-recount3" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> RNA-seq Analysis from recount3</h1>
<section id="introduction" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="introduction"><span class="header-section-number">1.1</span> Introduction</h2>
<p>RNA-seq allows the quantification of gene expression levels across the entire transcriptome and enables the identification of genes that are differentially expressed between biological conditions. Differential expression analysis is a key tool to understand molecular mechanisms underlying diseases.</p>
<p>Using edgeR for normalization and limma-voom for linear modeling, I aimed to identify genes that are differentially expressed between tumor and normal liver tissue and to look out for the global transcriptional differences between these two conditions.</p>
</section>
<section id="uploading-data" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="uploading-data"><span class="header-section-number">1.2</span> Uploading data</h2>
<section id="libraries" class="level3" data-number="1.2.1">
<h3 data-number="1.2.1" class="anchored" data-anchor-id="libraries"><span class="header-section-number">1.2.1</span> Libraries</h3>
<p>The specified task in the project is to perform an <em>RNA-seq</em> analysis using data from <code>recount3</code>. First I will load the necessary libraries and access the recount3 data.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"recount3"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"edgeR"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"limma"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"pheatmap"</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"RColorBrewer"</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="data-access" class="level3" data-number="1.2.2">
<h3 data-number="1.2.2" class="anchored" data-anchor-id="data-access"><span class="header-section-number">1.2.2</span> Data access</h3>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>human_projects <span class="ot">&lt;-</span> <span class="fu">available_projects</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>2026-02-23 07:44:31.425821 caching file sra.recount_project.MD.gz.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2026-02-23 07:44:32.024268 caching file gtex.recount_project.MD.gz.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2026-02-23 07:44:32.728762 caching file tcga.recount_project.MD.gz.</code></pre>
</div>
</div>
<p>Now I have to select a specific project (SRP<em>) to analyze and load its data. Thanks to the </em>metadata*, it is possible to search within the available projects to find one that matches the desired criteria.</p>
<p>Lets explore the different columns (metadata) on <code>human_projects</code> dataframe to understand what information is available to select a project for our analysis.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(human_projects)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "project"      "organism"     "file_source"  "project_home" "project_type"
[6] "n_samples"   </code></pre>
</div>
</div>
<p>Now we can properly filter projects…</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Search within the available projects (datafrmame notation instead of 'subset()')</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>sra_projects <span class="ot">&lt;-</span> human_projects[</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  human_projects<span class="sc">$</span>project_type <span class="sc">==</span> <span class="st">"data_sources"</span> <span class="sc">&amp;</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  human_projects<span class="sc">$</span>organism <span class="sc">==</span> <span class="st">"human"</span> <span class="sc">&amp;</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  human_projects<span class="sc">$</span>file_source <span class="sc">==</span> <span class="st">"sra"</span>,</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And look at the different projects found:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sra_projects)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>    project organism file_source     project_home project_type n_samples
1 SRP107565    human         sra data_sources/sra data_sources       216
2 SRP149665    human         sra data_sources/sra data_sources         4
3 SRP017465    human         sra data_sources/sra data_sources        23
4 SRP119165    human         sra data_sources/sra data_sources         6
5 SRP133965    human         sra data_sources/sra data_sources        12
6 SRP096765    human         sra data_sources/sra data_sources         7</code></pre>
</div>
</div>
<p>From what we learned in class, the selected data is going to depend on:</p>
<ol type="1">
<li><p>At least a boolean variable to be able to separate the samples into two different groups</p></li>
<li><p>Reasonable samples in both groups in order to achieve enough statistical power for the analysis</p></li>
</ol>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sra_projects[<span class="fu">order</span>(sra_projects<span class="sc">$</span>n_samples, <span class="at">decreasing =</span> <span class="cn">TRUE</span>),])</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>       project organism file_source     project_home project_type n_samples
3068 SRP166108    human         sra data_sources/sra data_sources      8464
5105 ERP107748    human         sra data_sources/sra data_sources      7598
310  SRP066781    human         sra data_sources/sra data_sources      3854
6731 SRP079058    human         sra data_sources/sra data_sources      3589
4458 ERP017126    human         sra data_sources/sra data_sources      3514
4024 ERP108653    human         sra data_sources/sra data_sources      2687</code></pre>
</div>
</div>
<p>The one with the most samples has an <span class="math inline">\(n = 8464\)</span>. This is a huge number and even though that is in theory a good thing, we want to avoid mashed up studies (which may have many different tissues, conditions and confounders). For the purpose of the project, I’ll be looking for a more manageable number of samples with an <span class="math inline">\(n\)</span> between 100 and 500, and with a clear variable of interest to separate the samples into two groups.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sra_projects[sra_projects<span class="sc">$</span>n_samples <span class="sc">&gt;=</span> <span class="dv">50</span> <span class="sc">&amp;</span> sra_projects<span class="sc">$</span>n_samples <span class="sc">&lt;=</span> <span class="dv">100</span>,], <span class="dv">20</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>      project organism file_source     project_home project_type n_samples
14  SRP125965    human         sra data_sources/sra data_sources        72
18  SRP057065    human         sra data_sources/sra data_sources        51
24  SRP157965    human         sra data_sources/sra data_sources        63
47  SRP201365    human         sra data_sources/sra data_sources        79
148 SRP065812    human         sra data_sources/sra data_sources        52
177 SRP144212    human         sra data_sources/sra data_sources        56
178 SRP119312    human         sra data_sources/sra data_sources        60
208 SRP109318    human         sra data_sources/sra data_sources        75
227 ERP104818    human         sra data_sources/sra data_sources        96
229 SRP132018    human         sra data_sources/sra data_sources        56
232 SRP055918    human         sra data_sources/sra data_sources        80
235 SRP114518    human         sra data_sources/sra data_sources        91
275 SRP066118    human         sra data_sources/sra data_sources        54
305 SRP074581    human         sra data_sources/sra data_sources        50
329 SRP111481    human         sra data_sources/sra data_sources        96
345 SRP061881    human         sra data_sources/sra data_sources        56
347 SRP109781    human         sra data_sources/sra data_sources        57
364 SRP058181    human         sra data_sources/sra data_sources        73
381 SRP081020    human         sra data_sources/sra data_sources        55
399 SRP049820    human         sra data_sources/sra data_sources        83</code></pre>
</div>
</div>
<p>After a bit of research, I decided to pick the project <code>SRP068976</code>, which has 99 samples and is titled “RNA-seq of 50 <a href="https://pubmed.ncbi.nlm.nih.gov/27119355/">paired hepatocellular carcinoma</a>”, according to <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE77314">NCBI’s <em>GEO</em></a>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>proj_info <span class="ot">&lt;-</span> sra_projects[sra_projects<span class="sc">$</span>project <span class="sc">==</span> <span class="st">"SRP068976"</span>,]</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>rse_srpSRP068976 <span class="ot">&lt;-</span> <span class="fu">create_rse</span>(proj_info)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>2026-02-23 07:44:38.10358 downloading and reading the metadata.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2026-02-23 07:44:39.097875 caching file sra.sra.SRP068976.MD.gz.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2026-02-23 07:44:39.84205 caching file sra.recount_project.SRP068976.MD.gz.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2026-02-23 07:44:40.410813 caching file sra.recount_qc.SRP068976.MD.gz.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2026-02-23 07:44:41.170985 caching file sra.recount_seq_qc.SRP068976.MD.gz.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2026-02-23 07:44:41.884079 caching file sra.recount_pred.SRP068976.MD.gz.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2026-02-23 07:44:41.983122 downloading and reading the feature information.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2026-02-23 07:44:42.600543 caching file human.gene_sums.G026.gtf.gz.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2026-02-23 07:44:43.987938 downloading and reading the counts: 99 samples across 63856 features.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2026-02-23 07:44:44.649146 caching file sra.gene_sums.SRP068976.G026.gz.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2026-02-23 07:44:46.09959 constructing the RangedSummarizedExperiment (rse) object.</code></pre>
</div>
</div>
</section>
</section>
<section id="exploration" class="level2" data-number="1.3">
<h2 data-number="1.3" class="anchored" data-anchor-id="exploration"><span class="header-section-number">1.3</span> Exploration</h2>
<section id="metadata" class="level3" data-number="1.3.1">
<h3 data-number="1.3.1" class="anchored" data-anchor-id="metadata"><span class="header-section-number">1.3.1</span> Metadata</h3>
<p>This are the general characteristics of the resulting RSE object:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>rse_srpSRP068976</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: RangedSummarizedExperiment 
dim: 63856 99 
metadata(8): time_created recount3_version ... annotation recount3_url
assays(1): raw_counts
rownames(63856): ENSG00000278704.1 ENSG00000277400.1 ...
  ENSG00000182484.15_PAR_Y ENSG00000227159.8_PAR_Y
rowData names(10): source type ... havana_gene tag
colnames(99): SRR3182261 SRR3129836 ... SRR3129932 SRR3129935
colData names(175): rail_id external_id ...
  recount_pred.curated.cell_line BigWigURL</code></pre>
</div>
</div>
</section>
<section id="format" class="level3" data-number="1.3.2">
<h3 data-number="1.3.2" class="anchored" data-anchor-id="format"><span class="header-section-number">1.3.2</span> Format</h3>
<p>Functions <code>compute_read_counts()</code> and <code>expand_sra_attributes</code> are used to get counts per read from the project and an insight of their different atributes (which is one metadata field)</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get counts per read</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(rse_srpSRP068976, <span class="st">"counts"</span>) <span class="ot">&lt;-</span> <span class="fu">compute_read_counts</span>(rse_srpSRP068976)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Get insight of the different attributes</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>rse_srpSRP068976 <span class="ot">&lt;-</span> <span class="fu">expand_sra_attributes</span>(rse_srpSRP068976)</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(rse_srpSRP068976)[, <span class="fu">grepl</span>(<span class="st">"^sra_attribute"</span>, <span class="fu">colnames</span>(<span class="fu">colData</span>(rse_srpSRP068976)))]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 99 rows and 3 columns
           sra_attribute.diagnosis sra_attribute.source_name
                       &lt;character&gt;               &lt;character&gt;
SRR3182261                  Normal                     Liver
SRR3129836                  Normal                     Liver
SRR3129837                  Normal                     Liver
SRR3129838                  Normal                     Liver
SRR3129839                  Normal                     Liver
...                            ...                       ...
SRR3129927                   Tumor                     Liver
SRR3129929                   Tumor                     Liver
SRR3129931                   Tumor                     Liver
SRR3129932                   Tumor                     Liver
SRR3129935                   Tumor                     Liver
           sra_attribute.tissue
                    &lt;character&gt;
SRR3182261                liver
SRR3129836                liver
SRR3129837                liver
SRR3129838                liver
SRR3129839                liver
...                         ...
SRR3129927                liver
SRR3129929                liver
SRR3129931                liver
SRR3129932                liver
SRR3129935                liver</code></pre>
</div>
</div>
<p><code>sra_attribute.diagnosis</code> has two different values: “Normal” and “Tumor”.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">colData</span>(rse_srpSRP068976)<span class="sc">$</span>sra_attribute.diagnosis)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Normal  Tumor 
    50     49 </code></pre>
</div>
</div>
<p>Among with this variable being just ideal to separate the samples, there is enough data in both groups to succesfully perform a differential expression analysis.</p>
<p>Something worth to mention is that, looking at the project on <em>GEO</em>, we can see that the samples are paired, which means that we have both a sample of normal tissue and a sample of tumor tissue for each patient.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">colData</span>(rse_srpSRP068976)<span class="sc">$</span>sra.sample_title, <span class="dv">20</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Patient 49 Normal" "Patient 1 Normal"  "Patient 2 Normal" 
 [4] "Patient 3 Normal"  "Patient 4 Normal"  "Patient 5 Normal" 
 [7] "Patient 6 Normal"  "Patient 7 Normal"  "Patient 10 Normal"
[10] "Patient 8 Normal"  "Patient 9 Normal"  "Patient 11 Normal"
[13] "Patient 12 Normal" "Patient 14 Normal" "Patient 13 Normal"
[16] "Patient 15 Normal" "Patient 16 Normal" "Patient 17 Normal"
[19] "Patient 18 Normal" "Patient 19 Normal"</code></pre>
</div>
</div>
</section>
</section>
<section id="preprocessing" class="level2" data-number="1.4">
<h2 data-number="1.4" class="anchored" data-anchor-id="preprocessing"><span class="header-section-number">1.4</span> Preprocessing</h2>
<section id="clustering-samples-into-their-id-groups" class="level3" data-number="1.4.1">
<h3 data-number="1.4.1" class="anchored" data-anchor-id="clustering-samples-into-their-id-groups"><span class="header-section-number">1.4.1</span> Clustering samples into their ID groups</h3>
<p>Before the analysis itself, I’m going to isolate the patient ID from the <code>sra.sample_title</code> attribute:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>rse_srpSRP068976<span class="sc">$</span>patient <span class="ot">&lt;-</span> <span class="fu">factor</span>(</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sub</span>(<span class="st">"Patient ([0-9]+) .*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>,</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colData</span>(rse_srpSRP068976)<span class="sc">$</span>sra.sample_title)</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(rse_srpSRP068976<span class="sc">$</span>patient)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 1 10 11 12 13 14 15 16 17 18 19  2 20 21 22 23 24 25 26 27 28 29  3 30 31 32 
 2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  1  2  2  2  2  2  2  2 
33 34 35 36 37 38 39  4 40 41 42 43 44 45 46 47 48 49  5 50  6  7  8  9 
 2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2 </code></pre>
</div>
</div>
<p>And, indeed, there are 50 patients with 2 samples each (except for the 26th). Until now is confirmed that the data is paired.</p>
</section>
<section id="basic-qc" class="level3" data-number="1.4.2">
<h3 data-number="1.4.2" class="anchored" data-anchor-id="basic-qc"><span class="header-section-number">1.4.2</span> Basic QC</h3>
<p>Basic <em>QC</em> is performed by looking at the proportion of assigned genes (genes that have at least one read assigned to them) over the total number of genes, that is, how many genes are useful and in which proportion.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>rse_srpSRP068976<span class="sc">$</span>assigned_gene_prop <span class="ot">&lt;-</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  rse_srpSRP068976<span class="sc">$</span>recount_qc.gene_fc_count_all.assigned <span class="sc">/</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  rse_srpSRP068976<span class="sc">$</span>recount_qc.gene_fc_count_all.total</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rse_srpSRP068976<span class="sc">$</span>assigned_gene_prop)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.5237  0.7621  0.7777  0.7705  0.7951  0.8169 </code></pre>
</div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(rse_srpSRP068976<span class="sc">$</span>assigned_gene_prop)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>As shown, the proportion of assigned genes is quite good, with a median of 77.77% and a mean of 77.05%. This means that most of the reads are being assigned to genes.</p>
<p>Box-plots are useful to determine whether the quality of the samples is independent of the diagnosis variable or not:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(rse_srpSRP068976)),</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> sra_attribute.diagnosis,</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> assigned_gene_prop)) <span class="sc">+</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Diagnosis"</span>) <span class="sc">+</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Assigned Gene Proportion"</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The quality of the samples does not seem to be confounded with the diagnosis variable.</p>
</section>
<section id="drop-lowly-expressed-genes" class="level3" data-number="1.4.3">
<h3 data-number="1.4.3" class="anchored" data-anchor-id="drop-lowly-expressed-genes"><span class="header-section-number">1.4.3</span> Drop lowly expressed genes</h3>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the original object to prevent recomputing</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>rse_srpSRP068976_unfiltered <span class="ot">&lt;-</span> rse_srpSRP068976</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out lowly expressed genes</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>gene_means <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">assay</span>(rse_srpSRP068976, <span class="st">"counts"</span>))</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(gene_means)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
0.000e+00 1.000e-02 8.000e-01 3.391e+02 5.293e+01 1.269e+06 </code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>rse_srpSRP068976 <span class="ot">&lt;-</span> rse_srpSRP068976[gene_means <span class="sc">&gt;</span> <span class="fl">0.1</span>, ]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p><code>model.matrix()</code> is used to create the design matrix for the analysis, which is based on the model used in class but includes both the patient ID and the diagnosis as variables.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span> patient <span class="sc">+</span> sra_attribute.diagnosis,</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> <span class="fu">colData</span>(rse_srpSRP068976)</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mod)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "(Intercept)"                  "patient10"                   
 [3] "patient11"                    "patient12"                   
 [5] "patient13"                    "patient14"                   
 [7] "patient15"                    "patient16"                   
 [9] "patient17"                    "patient18"                   
[11] "patient19"                    "patient2"                    
[13] "patient20"                    "patient21"                   
[15] "patient22"                    "patient23"                   
[17] "patient24"                    "patient25"                   
[19] "patient26"                    "patient27"                   
[21] "patient28"                    "patient29"                   
[23] "patient3"                     "patient30"                   
[25] "patient31"                    "patient32"                   
[27] "patient33"                    "patient34"                   
[29] "patient35"                    "patient36"                   
[31] "patient37"                    "patient38"                   
[33] "patient39"                    "patient4"                    
[35] "patient40"                    "patient41"                   
[37] "patient42"                    "patient43"                   
[39] "patient44"                    "patient45"                   
[41] "patient46"                    "patient47"                   
[43] "patient48"                    "patient49"                   
[45] "patient5"                     "patient50"                   
[47] "patient6"                     "patient7"                    
[49] "patient8"                     "patient9"                    
[51] "sra_attribute.diagnosisTumor"</code></pre>
</div>
</div>
<p>There are 49 patient coefficients (one for each patient except the reference) and one coefficient for the diagnosis variable, which is the most informative one…</p>
</section>
<section id="normalization" class="level3" data-number="1.4.4">
<h3 data-number="1.4.4" class="anchored" data-anchor-id="normalization"><span class="header-section-number">1.4.4</span> Normalization</h3>
<p>At this point, the data is ready for normalization and differential expression analysis. The <code>DGEList()</code> function from the <em>edgeR</em> package is used to create a DGEList object from the counts and the gene information, and then <code>calcNormFactors()</code> to compute the normalization factors.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">counts =</span> <span class="fu">assay</span>(rse_srpSRP068976, <span class="st">"counts"</span>),</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes =</span> <span class="fu">rowData</span>(rse_srpSRP068976)</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(dge)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p><code>voom()</code> function from the <em>limma</em> package estimates the mean-variance relationship and produces a transformed expression matrix that can be used for linear modeling.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>vGene <span class="ot">&lt;-</span> <span class="fu">voom</span>(dge, mod, <span class="at">plot =</span> <span class="cn">TRUE</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>eb_results <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(<span class="fu">lmFit</span>(vGene))</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>de_results <span class="ot">&lt;-</span> <span class="fu">topTable</span>(</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>  eb_results,</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">coef =</span> <span class="st">"sra_attribute.diagnosisTumor"</span>,</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">number =</span> <span class="cn">Inf</span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(de_results); <span class="fu">head</span>(de_results)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 41003    16</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>                   source type bp_length phase            gene_id
ENSG00000134690.10 HAVANA gene      2469    NA ENSG00000134690.10
ENSG00000100526.19 HAVANA gene      1836    NA ENSG00000100526.19
ENSG00000237649.7  HAVANA gene      3345    NA  ENSG00000237649.7
ENSG00000112984.11 HAVANA gene      3958    NA ENSG00000112984.11
ENSG00000143228.12 HAVANA gene      3362    NA ENSG00000143228.12
ENSG00000164611.12 HAVANA gene      2013    NA ENSG00000164611.12
                        gene_type gene_name level          havana_gene  tag
ENSG00000134690.10 protein_coding     CDCA8     2 OTTHUMG00000004320.1 &lt;NA&gt;
ENSG00000100526.19 protein_coding     CDKN3     1 OTTHUMG00000140302.3 &lt;NA&gt;
ENSG00000237649.7  protein_coding     KIFC1     2 OTTHUMG00000031209.4 &lt;NA&gt;
ENSG00000112984.11 protein_coding    KIF20A     2 OTTHUMG00000129195.3 &lt;NA&gt;
ENSG00000143228.12 protein_coding      NUF2     1 OTTHUMG00000034275.3 &lt;NA&gt;
ENSG00000164611.12 protein_coding     PTTG1     2 OTTHUMG00000130328.5 &lt;NA&gt;
                      logFC   AveExpr        t      P.Value    adj.P.Val
ENSG00000134690.10 4.043978 1.2304986 28.16270 8.453577e-34 3.466220e-29
ENSG00000100526.19 4.625529 0.7976569 27.26575 4.287480e-33 8.789978e-29
ENSG00000237649.7  4.132802 2.0841661 25.92257 5.329825e-32 7.284627e-28
ENSG00000112984.11 4.044777 1.5456917 25.32955 1.680714e-31 1.722858e-27
ENSG00000143228.12 4.589461 0.8598324 25.01408 3.125257e-31 2.562898e-27
ENSG00000164611.12 4.181153 1.7461350 24.76174 5.157552e-31 3.003476e-27
                          B
ENSG00000134690.10 66.37806
ENSG00000100526.19 64.76176
ENSG00000237649.7  62.50351
ENSG00000112984.11 61.33286
ENSG00000143228.12 60.65325
ENSG00000164611.12 60.26603</code></pre>
</div>
</div>
<p>Now we have the results of the differential expression analysis, with the log fold change, average expression, t-statistic, p-value and adjusted p-value for each gene. We can filter the results to find the differentially expressed genes based on a threshold for the adjusted p-value.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(de_results<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
13008 27995 </code></pre>
</div>
</div>
<p>And, as well as in class, MA and a volcano plots can be maked to visualize the results:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(eb_results, <span class="at">coef =</span> <span class="st">"sra_attribute.diagnosisTumor"</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">volcanoplot</span>(eb_results,</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">coef =</span> <span class="st">"sra_attribute.diagnosisTumor"</span>,</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">highlight =</span> <span class="dv">10</span>,</span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">names =</span> de_results<span class="sc">$</span>gene_name)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-23-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Lastly, this are the top 50 differentially expressed genes:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>exprs_heatmap <span class="ot">&lt;-</span> vGene<span class="sc">$</span>E[<span class="fu">rank</span>(de_results<span class="sc">$</span>adj.P.Val) <span class="sc">&lt;=</span> <span class="dv">50</span>, ]</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(</span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colData</span>(rse_srpSRP068976)[, <span class="st">"sra_attribute.diagnosis"</span>, <span class="at">drop =</span> <span class="cn">FALSE</span>]</span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="st">"Diagnosis"</span></span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a>pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(</span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a>  exprs_heatmap,</span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,</span>
<span id="cb55-12"><a href="#cb55-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>,</span>
<span id="cb55-13"><a href="#cb55-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb55-14"><a href="#cb55-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb55-15"><a href="#cb55-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_col =</span> df,</span>
<span id="cb55-16"><a href="#cb55-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"row"</span></span>
<span id="cb55-17"><a href="#cb55-17" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Tumor and normal samples separate into distinct clusters, indicating that the strongest differentially expressed genes are sufficient to discriminate between the two biological conditions.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extraemos el grupo biológico</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>col.group <span class="ot">&lt;-</span> rse_srpSRP068976<span class="sc">$</span>sra_attribute.diagnosis</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Convertimos a factor (por seguridad)</span></span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>col.group <span class="ot">&lt;-</span> <span class="fu">factor</span>(col.group)</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Asignamos colores</span></span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(col.group) <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="fu">nlevels</span>(col.group), <span class="st">"Set1"</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in brewer.pal(nlevels(col.group), "Set1"): minimal value for n is 3, returning requested palette with 3 different levels</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convertimos a vector de colores</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>col.group <span class="ot">&lt;-</span> <span class="fu">as.character</span>(col.group)</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a><span class="co"># MDS</span></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(vGene<span class="sc">$</span>E,</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> rse_srpSRP068976<span class="sc">$</span>sra_attribute.diagnosis,</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> col.group)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>There is a clear separation between the two conditions in the MDS plot, which indicates that the global transcriptional variation is strongly associated with the diagnosis variable.</p>
</section>
</section>
<section id="resultados" class="level2" data-number="1.5">
<h2 data-number="1.5" class="anchored" data-anchor-id="resultados"><span class="header-section-number">1.5</span> Resultados</h2>
</section>
<section id="results" class="level2" data-number="1.6">
<h2 data-number="1.6" class="anchored" data-anchor-id="results"><span class="header-section-number">1.6</span> Results</h2>
<p>This <em>differential expression analysis</em> identified a large number of genes significantly altered between tumor and normal liver tissue (FDR &lt; 0.05). The MA and volcano plots shows that many genes exhibit both strong fold changes and high statistical significance, indicating some sort of transcriptional dysregulation in <em>hepatocellular carcinoma</em>. Following, the heatmap of the top 50 differentially expressed genes shows a clustering by diagnosis, suggesting tumor and normal samples have different expression <em>forms</em>. Moreover, when looking to the MDS plot, the way the two conditions are clustered reinforces that diagnosis is fundamental on global transcriptional variation.</p>
</section>
</section>
<section id="references" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> References</h1>
<div id="refs" role="list">

</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>