---
title: "RNA-seq Analysis from recount3"
author: "Mateo Jiménez Sotelo"
date: "February 2026"
format:
  html:
    toc: true
    number-sections: true
bibliography: ../references.bib
nocite: "@*"
---

::: {.text-left}

### RNA-seq course project {.unnumbered}
### LCG22 {.unnumbered}
### UNAM {.unnumbered}

---

**Dataset:** SRP068976  
**Platform:** recount3 

:::

# RNA-seq Analysis from recount3

## Introduction

RNA-seq allows the quantification of gene expression levels across the entire transcriptome and enables the identification of genes that are differentially expressed between biological conditions. Differential expression analysis is a key tool to understand molecular mechanisms underlying diseases.

Using edgeR for normalization and limma-voom for linear modeling, I aimed to identify genes that are differentially expressed between tumor and normal liver tissue and to look out for the global transcriptional differences between these two conditions.

## Uploading data

### Libraries

The specified task in the project is to perform an *RNA-seq* analysis using data from `recount3`. First I will load the necessary libraries and access the recount3 data.

```{r}
#| warning: false
#| output: false
library("recount3")
library("edgeR")
library("limma")
library("ggplot2")
library("pheatmap")
library("RColorBrewer")
```

### Data access

```{r}
human_projects <- available_projects()
```

Now I have to select a specific project (SRP*) to analyze and load its data. Thanks to the *metadata*, it is possible to search within the available projects to find one that matches the desired criteria.

Lets explore the different columns (metadata) on `human_projects` dataframe to understand what information is available to select a project for our analysis.

```{r}
colnames(human_projects)
```

Now we can properly filter projects...

```{r}
# Search within the available projects (datafrmame notation instead of 'subset()')
sra_projects <- human_projects[
  human_projects$project_type == "data_sources" &
  human_projects$organism == "human" &
  human_projects$file_source == "sra",
]
```

And look at the different projects found:

```{r}
head(sra_projects)
```

From what we learned in class, the selected data is going to depend on:

1. At least a boolean variable to be able to separate the samples into two different groups

2. Reasonable samples in both groups in order to achieve enough statistical power for the analysis

```{r}
head(sra_projects[order(sra_projects$n_samples, decreasing = TRUE),])
```

The one with the most samples has an $n = 8464$. This is a huge number and even though that is in theory a good thing, we want to avoid mashed up studies (which may have many different tissues, conditions and confounders).
For the purpose  of the project, I'll be looking for a more manageable number of samples with an $n$ between 100 and 500, and with a clear variable of interest to separate the samples into two groups.

```{r}
head(sra_projects[sra_projects$n_samples >= 50 & sra_projects$n_samples <= 100,], 20)
```

After a bit of research, I decided to pick the project `SRP068976`, which has 99 samples and is titled "RNA-seq of 50 [paired hepatocellular carcinoma](https://pubmed.ncbi.nlm.nih.gov/27119355/)", according to [NCBI's *GEO*](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE77314).

```{r}
proj_info <- sra_projects[sra_projects$project == "SRP068976",]

rse_srpSRP068976 <- create_rse(proj_info)
```

## Exploration

### Metadata

This are the general characteristics of the resulting RSE object:

```{r}
rse_srpSRP068976
```

### Format

Functions `compute_read_counts()` and `expand_sra_attributes` are used to get counts per read from the project and an insight of their different atributes (which is one metadata field)

```{r}
# Get counts per read
assay(rse_srpSRP068976, "counts") <- compute_read_counts(rse_srpSRP068976)

# Get insight of the different attributes
rse_srpSRP068976 <- expand_sra_attributes(rse_srpSRP068976)
colData(rse_srpSRP068976)[, grepl("^sra_attribute", colnames(colData(rse_srpSRP068976)))]
```


`sra_attribute.diagnosis` has two different values: "Normal" and "Tumor".

```{r}
table(colData(rse_srpSRP068976)$sra_attribute.diagnosis)
```


Among with this variable being just ideal to separate the samples, there is enough data in both groups to succesfully perform a differential expression analysis.

Something worth to mention is that, looking at the project on *GEO*, we can see that the samples are paired, which means that we have both a sample of normal tissue and a sample of tumor tissue for each patient.

```{r}
head(colData(rse_srpSRP068976)$sra.sample_title, 20)
```

## Preprocessing

### Clustering samples into their ID groups

Before the analysis itself, I'm going to isolate the patient ID from the `sra.sample_title` attribute:

```{r}
rse_srpSRP068976$patient <- factor(
  sub("Patient ([0-9]+) .*", "\\1",
      colData(rse_srpSRP068976)$sra.sample_title)
)

table(rse_srpSRP068976$patient)
```


And, indeed, there are 50 patients with 2 samples each (except for the 26th). Until now is confirmed that the data is paired.

### Basic QC

Basic *QC* is performed by looking at the proportion of assigned genes (genes that have at least one read assigned to them) over the total number of genes, that is, how many genes are useful and in which proportion.

```{r}
rse_srpSRP068976$assigned_gene_prop <-
  rse_srpSRP068976$recount_qc.gene_fc_count_all.assigned /
  rse_srpSRP068976$recount_qc.gene_fc_count_all.total

summary(rse_srpSRP068976$assigned_gene_prop)
```

```{r}
hist(rse_srpSRP068976$assigned_gene_prop)
```

As shown, the proportion of assigned genes is quite good, with a median of 77.77% and a mean of 77.05%. This means that most of the reads are being assigned to genes.

Box-plots are useful to determine whether the quality of the samples is independent of the diagnosis variable or not:

```{r}
ggplot(as.data.frame(colData(rse_srpSRP068976)),
       aes(x = sra_attribute.diagnosis,
           y = assigned_gene_prop)) +
  geom_boxplot() +
  theme_bw() +
  xlab("Diagnosis") +
  ylab("Assigned Gene Proportion")
```

The quality of the samples does not seem to be confounded with the diagnosis variable.

### Drop lowly expressed genes

```{r}
# Save the original object to prevent recomputing
rse_srpSRP068976_unfiltered <- rse_srpSRP068976

# Filter out lowly expressed genes
gene_means <- rowMeans(assay(rse_srpSRP068976, "counts"))
summary(gene_means)

rse_srpSRP068976 <- rse_srpSRP068976[gene_means > 0.1, ]
```

`model.matrix()` is used to create the design matrix for the analysis, which is based on the model used in class but includes both the patient ID and the diagnosis as variables.

```{r}
mod <- model.matrix(
  ~ patient + sra_attribute.diagnosis,
  data = colData(rse_srpSRP068976)
)
colnames(mod)
```

There are 49 patient coefficients (one for each patient except the reference) and one coefficient for the diagnosis variable, which is the most informative one...

### Normalization

At this point, the data is ready for normalization and differential expression analysis. The `DGEList()` function from the *edgeR* package is used to create a DGEList object from the counts and the gene information, and then `calcNormFactors()` to compute the normalization factors.

```{r}
dge <- DGEList(
  counts = assay(rse_srpSRP068976, "counts"),
  genes = rowData(rse_srpSRP068976)
)
dge <- calcNormFactors(dge)
```

`voom()` function from the *limma* package estimates the mean-variance relationship and produces a transformed expression matrix that can be used for linear modeling.

```{r}
vGene <- voom(dge, mod, plot = TRUE)
```

```{r}
eb_results <- eBayes(lmFit(vGene))

de_results <- topTable(
  eb_results,
  coef = "sra_attribute.diagnosisTumor",
  number = Inf
)
dim(de_results); head(de_results)
```

Now we have the results of the differential expression analysis, with the log fold change, average expression, t-statistic, p-value and adjusted p-value for each gene. We can filter the results to find the differentially expressed genes based on a threshold for the adjusted p-value.

```{r}
table(de_results$adj.P.Val < 0.05)
```

And, as well as in class, MA and a volcano plots can be maked to visualize the results:

```{r}
plotMA(eb_results, coef = "sra_attribute.diagnosisTumor")

volcanoplot(eb_results,
            coef = "sra_attribute.diagnosisTumor",
            highlight = 10,
            names = de_results$gene_name)
```

Lastly, this are the top 50 differentially expressed genes:

```{r}
exprs_heatmap <- vGene$E[rank(de_results$adj.P.Val) <= 50, ]

df <- as.data.frame(
  colData(rse_srpSRP068976)[, "sra_attribute.diagnosis", drop = FALSE]
)

colnames(df) <- "Diagnosis"

pheatmap::pheatmap(
  exprs_heatmap,
  cluster_rows = TRUE,
  cluster_cols = TRUE,
  show_rownames = FALSE,
  show_colnames = FALSE,
  annotation_col = df,
  scale = "row"
)
```

Tumor and normal samples separate into distinct clusters, indicating that the strongest differentially expressed genes are sufficient to discriminate between the two biological conditions.

```{r}
# Extraemos el grupo biológico
col.group <- rse_srpSRP068976$sra_attribute.diagnosis

# Convertimos a factor (por seguridad)
col.group <- factor(col.group)

# Asignamos colores
levels(col.group) <- brewer.pal(nlevels(col.group), "Set1")

# Convertimos a vector de colores
col.group <- as.character(col.group)

# MDS
plotMDS(vGene$E,
        labels = rse_srpSRP068976$sra_attribute.diagnosis,
        col = col.group)
```

There is a clear separation between the two conditions in the MDS plot, which indicates that the global transcriptional variation is strongly associated with the diagnosis variable.

## Resultados

## Results

This *differential expression analysis* identified a large number of genes significantly altered between tumor and normal liver tissue (FDR < 0.05). The MA and volcano plots shows that many genes exhibit both strong fold changes and high statistical significance, indicating some sort of transcriptional dysregulation in *hepatocellular carcinoma*. Following, the heatmap of the top 50 differentially expressed genes shows a clustering by diagnosis, suggesting tumor and normal samples have different expression *forms*. Moreover, when looking to the MDS plot, the way the two conditions are clustered reinforces that diagnosis is fundamental on global transcriptional variation.

# References

::: {#refs}
:::
